### PR TITLE
[DROOLS-6707] move NamedEntryPoint and other classes out of drools-core

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6707

Related PRs:
https://github.com/kiegroup/drools/pull/4181
https://github.com/kiegroup/kogito-runtimes/pull/1971

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/impl/EntryPointsManager.java [hunk 1]
```diff
@@ -1,95 +1,13 @@
...
-                entryPoints.remove(removedNode.getEntryPoint().getEntryPointId());
-            }
-        }
-    }
 
-    public void reset() {
-        defaultEntryPoint.reset();
-        updateEntryPointsCache();
-    }
+public interface EntryPointsManager {
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/impl/EntryPointsManager.java]: Now it is an interface so what about moving it outside `impl` package?

File: drools-kiesession/src/main/java/org/drools/kiesession/entrypoints/NamedEntryPointsManager.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.drools.core.common.ReteEvaluator;
+import org.drools.core.impl.EntryPointsManager;
+import org.drools.core.impl.RuleBase;
+import org.drools.core.reteoo.EntryPointNode;
+import org.drools.core.reteoo.RuntimeComponentFactory;
+import org.drools.core.rule.EntryPointId;
+
+public class NamedEntryPointsManager implements EntryPointsManager {
+
+    private final ReteEvaluator reteEvaluator;
```
Comment (danielezonca) [inline drools-kiesession/src/main/java/org/drools/kiesession/entrypoints/NamedEntryPointsManager.java:34]: Isn't this class kiesession specific?

File: drools-traits/src/main/java/org/drools/traits/core/reteoo/TraitRuntimeComponentFactoryImpl.java [hunk 3]
```diff
@@ -62,8 +63,8 @@ public FactHandleFactory getFactHandleFactoryService() {
     }
 
     @Override
-    public NamedEntryPointFactory getNamedEntryPointFactory() {
-        return new TraitNamedEntryPointFactory();
+    public EntryPointFactory getNamedEntryPointFactory() {
```
Comment (danielezonca) [inline drools-traits/src/main/java/org/drools/traits/core/reteoo/TraitRuntimeComponentFactoryImpl.java]: Please align method name `getEntryPointFactory`

File: drools-kiesession/src/main/java/org/drools/kiesession/session/StatefulKnowledgeSessionImpl.java [hunk 4]
```diff
@@ -313,7 +313,7 @@ private StatefulKnowledgeSessionImpl(final long id,
         this.propagationIdCounter = new AtomicLong(propagationContext);
         init( config, environment, propagationContext );
 
-        this.entryPointsManager = new EntryPointsManager(this);
+        this.entryPointsManager = new NamedEntryPointsManager(this);
```
Comment (danielezonca) [inline drools-kiesession/src/main/java/org/drools/kiesession/session/StatefulKnowledgeSessionImpl.java:316]: Please use the factory every time you need to get a `*EntryPointsManager` instance

File: drools-core/src/main/java/org/drools/core/impl/RuleUnitExecutorImpl.java [hunk 5]
```diff
@@ -94,7 +94,7 @@ public RuleUnitExecutorImpl(RuleBase knowledgeBase, SessionConfiguration session
         this.nodeMemories = new ConcurrentNodeMemories(ruleBase);
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/impl/RuleUnitExecutorImpl.java:98]: What about moving this class to `drools-ruleunit-impl`?

File: drools-kiesession/src/main/java/org/drools/kiesession/session/StatefulKnowledgeSessionImpl.java (hunk 4 above)
Comment (mariofusco) [reply drools-kiesession/src/main/java/org/drools/kiesession/session/StatefulKnowledgeSessionImpl.java:316]: The `StatefulKnowledgeSessionImpl` is using methods specific of the `NamedEntryPointsManager` class, not exposed in the `EntryPointsManager` interface. If I use the factory method I'll have to downcast to the impl anyway and at this point I prefer to keep directly invoking the constructor.

File: drools-kiesession/src/main/java/org/drools/kiesession/entrypoints/NamedEntryPointsManager.java (hunk 2 above)
Comment (mariofusco) [reply drools-kiesession/src/main/java/org/drools/kiesession/entrypoints/NamedEntryPointsManager.java:34]: Done.

File: drools-traits/src/main/java/org/drools/traits/core/reteoo/TraitRuntimeComponentFactoryImpl.java (hunk 3 above)
Comment (mariofusco) [reply drools-traits/src/main/java/org/drools/traits/core/reteoo/TraitRuntimeComponentFactoryImpl.java]: Done.

File: drools-core/src/main/java/org/drools/core/impl/RuleUnitExecutorImpl.java (hunk 5 above)
Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/impl/RuleUnitExecutorImpl.java:98]: Done.

**Comment by kie-ci1** [comment]
**(tests) - drools job** #244 was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 20517
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.drools/244/testReport/org.kie.dmn.feel.client.showcase/FEELShowcaseIT/org.kie.dmn.feel.client.showcase.FEELShowcaseIT/">org.kie.dmn.feel.client.showcase.FEELShowcaseIT.org.kie.dmn.feel.client.showcase.FEELShowcaseIT</a></summary>
io.github.bonigarcia.wdm.WebDriverManagerException: Error HTTP 404 executing GET http://npm.taobao.org/mirrors/geckodriver HTTP/1.1
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.drools/244/display/redirect

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** #245 was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 432
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/245/testReport/org.acme.travel.tests.messaging.quarkus/MessagingIT/testProcess/">org.acme.travel.tests.messaging.quarkus.MessagingIT.testProcess</a></summary>
org.opentest4j.AssertionFailedError: expected: <0> but was: <3><br/>	at org.acme.travel.tests.messaging.quarkus.MessagingIT.testProcess(MessagingIT.java:110)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/245/display/redirect

**Pull request merged**